packages/ui/react-ui-form/src/components/Form/RefField.tsx

```diff
@@ -66,6 +72,54 @@ export const RefField = ({
   );
   const { options: availableOptions, loading: _loading } = useQueryRefOptions({ refTypeInfo, onQueryRefOptions });
 
+  const [showForm, setShowForm] = useState(false);
```

Comment:
perhaps time to rename this component, and promoted it to its own folder?
it's not quite a general purpose ComboBox, but "RefField" isn't an intuitive name to me.

ObjectSelector?
ObjectPicker?

e.g., it's not necessarily just for setting a "field" since we want to use it to select objects to insert into other components (board). 

i think perhaps we can further dry this out and move it into a new package along with QueryEditor (which should also be renamed). OR we move QueryEditor here? if we think these are all just "form" components?